docs/latest/custom-tools/choosing-a-base-class.md

```diff
@@ -0,0 +1,15 @@
+## Choosing A Base Class {#choosing-a-base-class}
+
+There are 3 base classes to choose from when building a tool:
```

Comment:
While we see it as a "base class", implementers might understand this better if we refer to them as different tool types. Each comes with different hooks/callbacks, and afford different default functionality to make creating different types of tools easier. A table of the callbacks the tool type offers, how to extend/use/implement it, and when/why you would use that tool type are all useful bits of information

Reply:
Yeah the semantics could use a little work, maybe we should call these 'Base Tools' colloquially?